presto-main/src/main/java/com/facebook/presto/metadata/SessionPropertyManager.java

```diff
@@ -155,9 +213,18 @@ private void updateWorkerSessionProperties()
 
     private List<PropertyMetadata<?>> getWorkerSessionProperties()
     {
-        return systemSessionPropertyProviders.values().stream()
-                .flatMap(manager -> manager.getSessionProperties().stream())
-                .collect(toImmutableList());
+        // Filter out java-worker related properties if native execution and coordinator sidecar is enabled
```

Comment:
Coming back to my comments on the previous commits, instead of doing this, can we simply not register the Java worker session properties if `nativeExecution` and `coordinatorSidecar` are enabled. That way we don't need to filter here.